esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -1080,6 +1080,7 @@ void EsdlServiceImpl::handleServiceRequest(IEspContext &context,
             {
                 m_pEsdlTransformer->process(context, EsdlResponseMode, srvdef.queryName(), mthdef.queryName(), out, origResp.str(), txResultFlags, ns, schema_location);
                 runPostEsdlScript(context, scriptContext, srvdef, mthdef, out, txResultFlags, ns, schema_location);
+                logdata.set("<LogDatasets/>");
```

Comment:
I think this is good for now.  I do wonder if in the future we will need to have the service script able to populate LogDatasets.  I guess we could already do it another way and we should worry about it in the future.  One scenario I can already imagine is where the service script is calling multiple roxie queries that each might want to log something.

Anyway, just something to keep in mind for the future.

Reply:
Even with this change though, the script type methods will still be able to populate LogDatasets in the PreLogging entry point, correct? It does appear that that the handleResultLogging() function is called for those also. If I'm reading correctly, this change just ensures that there is a LogDatasets context section to write into. 

I have seen other cases where a context section doesn't exist unless you define an entry point script that would operate on that section (eg EsdlResponse). So perhaps in a future change we'll move this change into a central location that ensures all context sections exist regardless of the execution path.

In any case this looks ok.

Reply:
Exactly, going forward without LogDatasets tag causes some errors into handleResultLogging(). It would be good to have this previously populated in a central location.